gulp-tasks/build-packages.js

```diff
@@ -68,14 +84,98 @@ const buildPackage = (packagePath, buildType) => {
 
   const plugins = rollupHelper.getDefaultPlugins(buildType);
 
-  // This makes Rollup assume workbox-core will be added to the global
+  // This makes Rollup assume workbox-* will be added to the global
   // scope and replace references with the core namespace
-  const globals = {
-    'workbox-core': `${constants.NAMESPACE_PREFIX}.core`,
+  const globals = (moduleId) => {
+    // This regex matches for (workbox-*)(/any/path/here.mjs)
+    const workboxModuleIdRegex = /(workbox-\w*)([/\w.]*)*/g;
+    const result = workboxModuleIdRegex.exec(moduleId);
+    if (!result) {
+      throw new Error(`Unknown global module ID: ${moduleId}`);
+    }
+
+    const packageName = result[1];
+    let importFilePath = null;
+    if (result.length === 3) {
+      importFilePath = result[2];
+      if (importFilePath && importFilePath.indexOf('/') === 0) {
+        importFilePath = importFilePath.replace('/', '');
+      }
+    }
+
+    // Get a packages browser namespace so we know where it will be
+    // ont he global scope (i.e. google.workbox.????)
+    let browserNamespace = null;
+    const packagePath = path.join(__dirname, '..', 'packages', packageName);
+    try {
+      const pkg = require(path.join(packagePath, 'package.json'));
+      browserNamespace = pkg.workbox.browserNamespace;
+    } catch (err) {
+      logHelper.error(`Unable to get browserNamespace for package: ` +
+        `'${packageName}'`);
+      logHelper.error(err);
+      throw err;
+    }
+
+    let globalNamespace = `${constants.NAMESPACE_PREFIX}.${browserNamespace}`;
+    let fileNamespace = '';
+    // If the module pulls in a specific files we'll need to add this
```

Comment:
Check the grammar. I'm not sure if you want to say "a specific file" + "it" or "specific files" + "them".